### ISSUE TITLE
Entities not created in 2023.9.0b2

### ISSUE THREAD

**Issue by dof250** [jcgoette/baby_buddy_homeassistant#127]
I had to remove the babybuddy integration and tried to re-add it. When re-adding it now shows the errors below in 2023.9.0. It only creates the 4 "select" entities now. (diaper_color, diaper_type, feeding_method and feeding_type)

All other entities are missing and not recreated. Below the errors in the log:

babybuddy: Not adding entity with invalid device info: Invalid device info {'configuration_url': 'https://URL:443/children/CHILD/dashboard/', 'default_name': 'Baby', 'identifiers': {('babybuddy', 1)}} for 'babybuddy' config entry: device info needs to either describe a device, link to existing device or provide extra information.

babybuddy: Not adding entity with invalid device info: Invalid device info {'default_name': 'Baby', 'identifiers': {('babybuddy', 1)}} for 'babybuddy' config entry: device info needs to either describe a device, link to existing device or provide extra information.

Any help would be great.

**Comment by jcgoette** [comment]
2023.9 isn't fully released yet?

**Comment by dof250** [comment]
That's right, but I don't think it's an issue with 2023.9. I think it has been there before the upgrade, but if you would want me to close it and wait for the final release of 2023.9 that's okay for me.

**Comment by jcgoette** [comment]
Yea, I can't support pre-releases. Feel free to send a PR or at least link a relevant release note.

**Comment by utsmok** [comment]
Same issue here after installing the beta release candidate version 2023.9, not sure if its a bug or a caused by a change in sensor support. The only change that seems relevant for now is: https://developers.home-assistant.io/blog/2023/08/23/sensor-updates/.

Here are some related errors from the log of my home assistance instance:


Logger: homeassistant.components.switch
Source: helpers/entity_platform.py:621
Integration: Switch (documentation, issues)
First occurred: 14:16:03 (6 occurrences)
Last logged: 14:31:04

babybuddy: Not adding entity with invalid device info: Invalid device info {'default_name': '[name hidden]', 'identifiers': {('babybuddy', 1)}} for 'babybuddy' config entry: device info needs to either describe a device, link to existing device or provide extra information.




Logger: custom_components.babybuddy
Source: helpers/update_coordinator.py:235
Integration: Baby Buddy (documentation, issues)
First occurred: 14:29:58 (1 occurrences)
Last logged: 14:29:58

Error fetching babybuddy data: Cannot connect to host homeassistant.local:8000 ssl:default [Invalid argument]




Logger: homeassistant.components.sensor
Source: helpers/entity_platform.py:621
Integration: Sensor (documentation, issues)
First occurred: 14:16:03 (66 occurrences)
Last logged: 14:31:04

babybuddy: Not adding entity with invalid device info: Invalid device info {'configuration_url': 'http://homeassistant.local:8000/children/'[name hidden]'/dashboard/', 'default_name': 'Baby '[name hidden]', 'identifiers': {('babybuddy', 1)}} for 'babybuddy' config entry: device info needs to either describe a device, link to existing device or provide extra information.

babybuddy: Not adding entity with invalid device info: Invalid device info {'configuration_url': 'http://192.168.2.23:8000/children/'[name hidden]'/dashboard/', 'default_name': 'Baby '[name hidden]'', 'identifiers': {('babybuddy', 1)}} for 'babybuddy' config entry: device info needs to either describe a device, link to existing device or provide extra information.

**Comment by jcgoette** [comment]
Not seeing anything under [breaking changes](https://rc.home-assistant.io/blog/2023/08/30/release-20239/#breaking-changes), but that doesn't mean a lot. 

If this ends up breaking with official release and I haven't sorted the device issue by then, I'll just remove device support.

**Comment by dbdmdbdmdbdm** [comment]
@jcgoette how do we resolve this issue? do we remove the default name mentioned in the other issue? are these related?

**Comment by pjrt** [comment]
I am having this issue. 2023.9 is now officially out. 

It think this issue needs to be re-opened, if newer issues are being closed and pointing to this one.

**Comment by seanhagen** [comment]
I'm getting the same error:

```
2023-09-06 22:40:23.536 ERROR (MainThread) [homeassistant.components.sensor] babybuddy: Not adding entity 
with invalid device info: Invalid device info {'configuration_url': 'http://10.100.0.10:8000/children/child-1/dashboard/', 
'default_name': 'Child 1', 'identifiers': {('babybuddy', 1)}} for 'babybuddy' config entry: device 
info needs to either describe a device, link to existing device or provide extra information.
```

This has broken automations I had set up, and now I've got an angry partner wondering when the zigbee buttons I set up will be working again 😅

**Comment by pjrt** [comment]
It seems this PR fixes it. https://github.com/jcgoette/baby_buddy_homeassistant/pull/131

**Comment by socbrian** [comment]
Can we re-open this till the PR is merged?

**Comment by jcgoette** [comment]
Closed by #131.
